src/abstractions/IRequestAdapter.cs

```diff
@@ -86,6 +86,6 @@ public interface IRequestAdapter
         /// <param name="requestInfo">The RequestInformation object to use for the HTTP request.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use for cancelling the requests.</param>
         /// <returns>The native HTTP request.</returns>
-        Task<T?> ConvertToNativeRequestAsync<T>(RequestInformation requestInfo, CancellationToken cancellationToken = default);
```

Comment:
I think we could probably revert this for now if we can. As `IRequestAdapter` is a public interface, an alternative implementation (not necessarily implemented by us) may choose to return the null type as much as our implementation here does not. 
Also tagging @baywet to weigh in here.

Reply:
👍 👍 
Changes reverted in the meantime

Reply:
In general I'd rather avoid changing the interfaces here, so thank you for reverting. In this specific instance, I'm not sure "allowing" implementers to return null makes sense. They probably should return something or throw to avoid derailing callers. Andrew: do you remember the context here?

Reply:
I think this came in as part of the of the NRT work with all other methods in the interface returning nullable types. At that time the repos were separate and this may be clearer now with the monorepo with the implementation and the interface together. If this was added today, we'd probably not do this as you suggest.